### DISCUSSION TITLE
What will be the suitable version for @aws-sdk/client-sqs to run sqs-consumer ?

### DISCUSSION THREAD

**Discussion by chronos25** [bbc/sqs-consumer#455] [Help] [answered]
### Summary

I am encountering multiple errors on my machine with different versions of the @aws-sdk/client-sqs and sqs-consumer packages. If I am using the latest versions of these packages I'm getting error for them as well. 

```

"@aws-sdk/client-sqs": "3.501.0" and "sqs-consumer": "^5.8.0"  its throwing - SQS receive message failed: this.sqs.receiveMessage is not a function.

"@aws-sdk/client-sqs": "3.32.0" and "sqs-consumer": "^5.8.0"  its throwing - SQS receive message failed: this.sqs.receiveMessage is not a function.

"@aws-sdk/client-sqs": "3.501.0" and "sqs-consumer": "^8.1.5" its throwing - SQS receive message failed: Unexpected non-whitespace character after JSON at position 4 (line 1 column 5)
  Deserialization error: to see the raw response, inspect the hidden field {error}.$response on this object.
```
Looks like there is issue with version of both package. Can somebody provide me the correct version of both packages.

Thanks </>

**Comment by nicholasgriffintn** [comment, marked as answer]
Hey, 5.8.0 only works with AWS SDK V2, it's specifically carved out only for that version of the AWS SDK as there are major differences between that and V3.

For the latest version of the AWS SDK (V3), you'll need to use the latest version of sqs-consumer.

It looks like you are for some reason loading two versions of sqs-consumer? You only need one. The latter error with the latest versions looks to be potentially the new JSON protocol in v8 and the latest versions of the AWS SDK, for that, you need to match the AWS SDK version across your system, or use a version prior to V8 of sqs-consumer.

Hopefully that answers your question :).
